src/app/components/GoogleSheetsInput.svelte

```diff
@@ -0,0 +1,172 @@
+<script>
+  import worker from "../spawn-worker";
+
+  import InputField from "./InputField.svelte";
+  import Button from "./Button.svelte";
+
+  var error = null;
+  let googleSheetsURL = "";
+
+  // Client ID and API key from the Developer Console
+  const CLIENT_ID = process.env.G_CLIENT_ID;
+  const API_KEY = process.env.G_API_KEY;
+
+  // Array of API discovery doc URLs for APIs used by the quickstart
+  const DISCOVERY_DOCS = [
+    "https://sheets.googleapis.com/$discovery/rest?version=v4",
+  ];
+
+  // Authorization scopes required by the API; multiple scopes can be
+  // included, separated by spaces.
+  const SCOPES = "https://www.googleapis.com/auth/spreadsheets.readonly";
+
+  /**
+   *  On load, called to load the auth2 library and API client library.
+   */
+  function handleClientLoad() {
+    gapi.load("client:auth2", initClient);
+  }
+
+  /**
+   *  Initializes the API client library and sets up sign-in state
+   *  listeners.
+   */
+  function initClient() {
+    error = null;
+    gapi.client
+      .init({
+        apiKey: API_KEY,
+        clientId: CLIENT_ID,
+        discoveryDocs: DISCOVERY_DOCS,
+        scope: SCOPES,
+      })
+      .then(
+        () => {
+          // Listen for sign-in state changes.
+          gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
+
+          // Handle the initial sign-in state.
+          updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
+        },
+        function () {
+          error = "Error: Could not connect to Google Sheets";
+        }
+      );
+  }
+
+  /**
+   *  Called when the signed in status changes, to update the UI
+   *  appropriately. After a sign-in, the API is called.
+   */
+  function updateSigninStatus(isSignedIn) {
+    if (isSignedIn) {
+      getValuesFromSpreadSheet();
+    } else {
+      handleAuthClick();
+      getValuesFromSpreadSheet();
+    }
+  }
+
+  /**
+   *  Sign in the user upon button click.
+   */
+  function handleAuthClick(event) {
```

Comment:
When possible, prefer function names that describe the _purpose_ of what they do, rather than `handleEVENT` names.

```suggestion
  function signInToGoogleAPI(event) {
```